pyrefly/lib/alt/answers.rs

```diff
@@ -79,6 +79,7 @@ pub struct Index {
 struct OverloadedCallee {
     all_overloads: Vec<Callable>,
     closest_overload: Callable,
+    closest_overload_type: Option<Arc<Type>>,
```

Comment:
I wonder if the reason we didn't store this the whole time was the memory consequence. but I guess the arc mitigates a lot of this